src/basics/Basics.php

```diff
@@ -0,0 +1,101 @@
+<?php
+/**
+ * Php version 8.1.9
+ * 
+ * @category No_Category
+ * @package  No_Package
+ * @author   Andrey Bakayev <andreybakayev@gmail.com>
+ * @license  https://github.com/AndriiBakayev free
+ * @link     https://github.com/AndriiBakayev
+ */
+namespace basics;
+
+/**
+ * Basics Class and implement the BasicsInterface and following methods:
+ * getMinuteQuarter(), isLeapYear(), isSumEqual()
+ * See details below.
+ * 
+ * @category No_Category
+ * @package  No_Package
+ * @author   Andrey Bakayev <andreybakayev@gmail.com>
+ * @license  https://github.com/AndriiBakayev free
+ * @link     https://github.com/AndriiBakayev
+ */
+class Basics implements BasicsInterface
+{
+    /**
+     * Stores validator object 
+     *
+     * @var BasicsValidatorInterface validator - private value for storing 
+     * validator object  of class ValidatorInterface, whitch used by all 
+     * functions to validate input parameter values for throwing 
+     * InvalidArgumentExceptionwhen the parameter values are out 
+     * of accepted bounds
+     */
+    private BasicsValidatorInterface $_Validator;
+
+    /**
+     * Constructor of the class that initialises private $validator variable
+     * acepts obgect of BasicsValidator class
+     *
+     * @param BasicsValidatorInterface $validator object of BasicsValidator class
+     *
+     * @return void
+     */
+    public function __construct(BasicsValidatorInterface $validator)
+    {
+        $this->_Validator = $validator;
+    }
+
+    /**
+     * GetMinuteQuarter returns the quarter of a hour according to parameter 
+     * $minute given
+     *
+     * @param int $minute is an integer value with a value between 00 an 59
+     *                    when it does, function returns:
+     * 
+     * @return string  one of the values: "first", "second", "third" and "fourth".
+     * when it, else it:
+     *
+     * @throws InvalidArgumentException if $minute is negative of greater then 59.
+     */
+    public function getMinuteQuarter(int $minute): string
+    {
+        $this->_Validator->isMinutesException($minute);
+        return $minute === 0 ? 'fourth'
+            : ($minute <= 15 ? 'first'
```

Comment:
please use if else instead of ?:
it is only usfull if you have 2 or maximum 3 checks